databricks/koalas/indexing.py

```diff
@@ -1696,6 +1696,21 @@ def _select_cols_else(
             )
 
     def __setitem__(self, key, value):
+        if is_list_like(value) and not isinstance(value, spark.Column):
+            iloc_item = self[key]
+            if not is_list_like(key) or not is_list_like(iloc_item):
+                raise ValueError("setting an array element with a sequence.")
+            elif is_list_like(key) and (len(iloc_item) != len(value)):
+                if self._is_series:
+                    raise ValueError(
+                        "cannot set using a list-like indexer with a different length than "
+                        "the value"
+                    )
+                else:
+                    raise ValueError(
+                        "shape mismatch: value array of shape ({},) could not be broadcast "
+                        "to indexing result of shape {}".format(len(value), iloc_item.shape)
```

Comment:
`iloc_item.shape` calls `len(self)` in it which is already called at line 1703. Can we avoid call it twice?

Reply:
Oh yeah, definitly.

Thanks for the review!